pyanaconda/modules/network/nm_client.py

```diff
@@ -492,41 +529,71 @@ def add_connection_from_ksdata(nm_client, network_data, device_name, activate=Fa
         ifname_option_values
     )
 
-    for con, device_name in connections:
-        log.debug("add connection: %s for %s\n%s", con.get_uuid(), device_name,
-                  con.to_dbus(NM.ConnectionSerializationFlags.NO_SECRETS))
+    for connection, device_name in connections:
+        log.debug("add connection (activate=%s): %s for %s\n%s",
+                  activate, connection.get_uuid(), device_name,
+                  connection.to_dbus(NM.ConnectionSerializationFlags.NO_SECRETS))
         device_to_activate = device_name if activate else None
-        nm_client.add_connection2(
-            con.to_dbus(NM.ConnectionSerializationFlags.ALL),
-            (NM.SettingsAddConnection2Flags.TO_DISK |
-             NM.SettingsAddConnection2Flags.BLOCK_AUTOCONNECT),
-            None,
-            False,
-            None,
-            _connection_added_cb,
+        add_and_activate_connection_sync(
```

Comment:
I don't really like the `and` word in the function name. Most of the time it means that the functions should be split to two.

Reply:
Why do you think this is the case ?
Also see: https://developer.gnome.org/libnm/stable/NMClient.html#nm-client-add-and-activate-connection-async

Reply:
Because smaller function is easier to call and simpler for modification. Also having:
```
add_and_activate_connection_sync()
```
is not giving you much benefit compared to
```
add_connection_sync()
activate_connection_sync()
```

You will strip dependencies to just the parts requiring it. And the last reason is that it's simpler for testing because it's smaller.

However, this is not something for blocking this PR. It's just note. Reason for block was the commit squashing.

Reply:
If you wanted to get pedantic, it's used exactly once so maybe mark it as _private. Also, "activate" could be dropped from name since it's in the parameter list. But I don't think it's important...

Reply:
I split the function:
https://github.com/rhinstaller/anaconda/pull/2861/commits/ce593eed2b4706fd82876276b315c6010d5b8079